replicator/src/core.rs

```diff
@@ -102,11 +105,16 @@ pub async fn start_replicator() -> anyhow::Result<()> {
     Ok(())
 }
 
-async fn init_state_store(
-    config: &ReplicatorConfig,
-) -> anyhow::Result<impl StateStore + Clone + Send + Sync + fmt::Debug + 'static> {
+async fn init_state_store(config: &ReplicatorConfig) -> anyhow::Result<BoxedStateStore> {
     match config.state_store {
-        StateStoreConfig::Memory => Ok(MemoryStateStore::new()),
+        StateStoreConfig::Memory => Ok(BoxedStateStore::new(MemoryStateStore::new())),
+        StateStoreConfig::Postgres => {
+            migrate_state_store(&config.source).await?;
+            Ok(BoxedStateStore::new(PostgresStateStore::new(
+                config.pipeline.id,
```

Comment:
I totally overlooked this while implementing the state store, we can easily keep the pipeline id within the struct implementing the state store since an instance is bound to a single pipeline. Thanks for doing this!